cwl/src/main/scala/cwl/ExpressionEvaluator.scala

```diff
@@ -24,6 +24,16 @@ object ExpressionEvaluator {
   type ECMAScriptFunction = String Refined MatchesRegex[ECMAScriptFunctionWitness.T]
   type MatchesECMAFunction = MatchesRegex[ECMAScriptFunctionWitness.T]
 
+  // This is not ECMAScript, just what CWL uses for interpolated expressions. A dollar sign immediately followed
+  // by a paren followed by anything that isn't a close paren followed by a close paren. Capturing groups around
+  // the fore, expression and after portions to concatenate fragments for evaluation. Obviously this will break
+  // with nested parens. It's not clear what the specification is here; this simplistic implementation may need
+  // to be revisited.
+  val InterpolatedStringWitness = Witness("""(?s)(.*)\$\(([^\)]+)\)(.*)""")
```

Comment:
Does this work for my [favorite test](https://github.com/common-workflow-language/common-workflow-language/blob/91f108df4d4ca567e567fc65f61feb0674467a84/v1.0/v1.0/search.cwl#L62)?

Reply:
Good catch! No it won't because that has multiple expressions in the string. I wasn't previously aware of anything like that in the wild. 😦  I'll update the code and test to handle > 1.